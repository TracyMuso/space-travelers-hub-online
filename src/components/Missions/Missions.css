#missions {
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 1rem;
}

#mission-table {
  width: 90%;
  border-collapse: collapse;
  border-spacing: 0;
  border: 1px solid #ddd;
}

tbody :nth-child(odd) {
  background-color: #f5f5f5;
}

#mission-table th,
#mission-table td {
  padding: 10px;
  border: 1px solid #ddd;
  text-align: left;
}
