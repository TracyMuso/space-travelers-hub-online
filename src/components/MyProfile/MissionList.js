import { useSelector } from 'react-redux';

const MissionList = () => {
  const missions = useSelector((state) => state.missionsReducer);

  const reservedTrue = {
    listStyle: 'none',
    display: 'flex',
    flexDirection: 'row',
  };

  const reservedFalse = {
    display: 'none',
  };

  return (
    <>
      {missions.map((item) => (
        <li key={item.id} style={item.reserved ? reservedTrue : reservedFalse}>
          {item.name}
        </li>
      ))}
    </>
  );
};

export default MissionList;
