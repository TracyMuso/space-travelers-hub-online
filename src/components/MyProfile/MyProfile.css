.profile-header-container {
  display: flex;
  flex-direction: row;
}

.profile-header-container > h2 {
  width: 50%;
  margin-left: 5%;
}

.profile-list-container {
  display: flex;
  flex-direction: row;
}

.profile-list-container > ul {
  width: 50%;
  margin-left: 5%;
  list-style: none;
  padding: 0;
}

.profile-list-container > ul > li {
  border: solid gray 1px;
  padding: 1em;
}
