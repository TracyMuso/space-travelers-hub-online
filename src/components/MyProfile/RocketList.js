import { useSelector } from 'react-redux';

const RocketList = () => {
  const state = useSelector((state) => state.rocketReducer);

  const reservedTrue = {
    listStyle: 'none',
    display: 'flex',
    flexDirection: 'row',
  };

  const reservedFalse = {
    display: 'none',
  };

  return (
    <>
      {state.map((item) => (
        <li key={item.id} style={item.reserved ? reservedTrue : reservedFalse}>
          {item.name}
        </li>
      ))}
    </>
  );
};

export default RocketList;
