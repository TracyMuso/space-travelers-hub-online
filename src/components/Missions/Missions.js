import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Button } from 'react-bootstrap';
import './Missions.css';
import { joinMission, fetchMissions, leaveMission } from '../../redux/mission/missions';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionsReducer);

  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  const handleJoin = (id) => {
    dispatch(joinMission(id));
  };

  const handleLeave = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <section id="missions">
      <table id="mission-table">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>{'  '}</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.id}>
              <td>{mission.name}</td>
              <td>{mission.description}</td>
              {mission.reserved ? (
                <>
                  <td>
                    <Button className="btn btn-success active">Active member</Button>
                  </td>
                  <td>
                    <Button className="btn btn-danger active" onClick={() => handleLeave(mission.id)}>Leave Mission</Button>
                  </td>
                </>
              ) : (
                <>
                  <td>
                    <Button className="btn btn-secondary active">Not a member</Button>
                  </td>
                  <td>
                    <Button className="btn btn-primary active" onClick={() => handleJoin(mission.id)}>Join Mission</Button>
                  </td>
                </>
              )}
            </tr>
          ))}
        </tbody>
      </table>
    </section>
  );
};

export default Missions;
