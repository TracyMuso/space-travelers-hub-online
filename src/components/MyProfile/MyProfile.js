import RocketList from './RocketList';
import MissionList from './MissionList';
import './MyProfile.css';

const MyProfile = () => (
  <>
    <div className="profile-header-container">
      <h2>My Missions</h2>
      <h2>My Rockets</h2>
    </div>
    <div className="profile-list-container">
      <ul>
        <MissionList />
      </ul>
      <ul>
        <RocketList />
      </ul>
    </div>
  </>
);

export default MyProfile;
