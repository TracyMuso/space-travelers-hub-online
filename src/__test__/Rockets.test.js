import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketsList, { Rocket } from '../components/Rockets/Rockets';
import MyProfile from '../components/MyProfile/MyProfile';
import Missions from '../components/Missions/Missions';
import state from '../__mocks__/mockState';

describe('Testing Snapshots', () => {
  test('Rockets Snapshot', () => {
    const snapDragon = renderer.create(
      <Provider store={store}>
        <RocketsList />
      </Provider>,
    );
    const tree = snapDragon.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('My Profile Snapshot', () => {
    const snapProfile = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    const tree = snapProfile.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Missions SnapShot', () => {
    const snapMissions = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const tree = snapMissions.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Rocket List Test', () => {
    const snapRocketList = renderer.create(
      <Provider store={store}>
        {state.map((rocket) => (
          <Rocket
            key={rocket.id}
            id={rocket.id}
            name={rocket.name}
            description={rocket.description}
            flickrImages={rocket.flickr_images}
            reserved={rocket.reserved}
          />
        ))}
      </Provider>,
    );
    const tree = snapRocketList.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
