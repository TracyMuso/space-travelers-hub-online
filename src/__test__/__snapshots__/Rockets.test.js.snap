// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshots Missions SnapShot 1`] = `
<section
  id="missions"
>
  <table
    id="mission-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
            
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;

exports[`Testing Snapshots My Profile Snapshot 1`] = `
Array [
  <div
    className="profile-header-container"
  >
    <h2>
      My Missions
    </h2>
    <h2>
      My Rockets
    </h2>
  </div>,
  <div
    className="profile-list-container"
  >
    <ul />
    <ul />
  </div>,
]
`;

exports[`Testing Snapshots Rocket List Test 1`] = `
Array [
  <li>
    <img
      alt="A rocket launching"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div>
      <h2>
        Falcon 1
      </h2>
      <p>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          Booked
        </span>
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
      </p>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "dodgerblue",
            "border": "0",
            "borderRadius": "5px",
            "color": "whitesmoke",
            "padding": "7px",
          }
        }
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </li>,
  <li>
    <img
      alt="A rocket launching"
      src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
    />
    <div>
      <h2>
        Falcon 9
      </h2>
      <p>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          Booked
        </span>
        Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
      </p>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "dodgerblue",
            "border": "0",
            "borderRadius": "5px",
            "color": "whitesmoke",
            "padding": "7px",
          }
        }
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </li>,
  <li>
    <img
      alt="A rocket launching"
      src="https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg"
    />
    <div>
      <h2>
        Falcon Heavy
      </h2>
      <p>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          Booked
        </span>
        With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
      </p>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "dodgerblue",
            "border": "0",
            "borderRadius": "5px",
            "color": "whitesmoke",
            "padding": "7px",
          }
        }
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </li>,
  <li>
    <img
      alt="A rocket launching"
      src="https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg"
    />
    <div>
      <h2>
        Starship
      </h2>
      <p>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          Booked
        </span>
        Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
      </p>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "dodgerblue",
            "border": "0",
            "borderRadius": "5px",
            "color": "whitesmoke",
            "padding": "7px",
          }
        }
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </li>,
]
`;

exports[`Testing Snapshots Rockets Snapshot 1`] = `
<ul
  className="rocket-list-container"
/>
`;
